### PR TITLE
👍 add fixup actions to gin-log (fixup, amend, reword)

### PR DESCRIPTION
**Pull request by atusy**
adds

- `<Plug>(gin-action-fixup)`
- `<Plug>(gin-action-fixup:fixup)`
- `<Plug>(gin-action-fixup:amend)`
- `<Plug>(gin-action-fixup:reword)`

## Summary by CodeRabbit

- **New Features**
  - Introduced new commands for advanced "fixup" operations within text buffers.
  - Enhanced version control interaction with new fixup and amend capabilities.

- **Improvements**
  - Streamlined the process of gathering and processing commit candidates for fixup actions.

- **Documentation**
  - Updated user instructions to accommodate new fixup and amend command workflows.

### CODE REVIEW

**Comment by lambdalisue** [comment]
Additionally, I've fixed CI so please rebase it.
https://github.com/lambdalisue/gin.vim/pull/115

File: denops/gin/action/fixup.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+  gatherCandidates: GatherCandidates<Candidate>,
+): Promise<void> {
+  const xs = await gatherCandidates(denops, bufnr, range);
+  const commit = xs.map((v) => v.commit).join("\n");
+  const target = kind === "fixup" ? commit : `${kind}:${commit}`;
+  // Do not block Vim so that users can edit commit message
+  await denops.call("denops#notify", "gin", "command", ["", [
+    "commit",
+    `--fixup=${target}`,
+  ]]);
```
[the remark this answers is not in the document]
Comment (atusy) [reply denops/gin/action/fixup.ts]: Thanks. Seems working.

**Comment by atusy** [comment]
Thank you for the review and fixing CI.
Rebased with the suggested code.

File: denops/gin/action/fixup.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  kind: "fixup" | "amend" | "reword",
+  gatherCandidates: GatherCandidates<Candidate>,
+): Promise<void> {
+  const xs = await gatherCandidates(denops, bufnr, range);
+  const commit = xs.map((v) => v.commit).join("\n");
+  const target = kind === "fixup" ? commit : `${kind}:${commit}`;
+  // Do not block Vim so that users can edit commit message
+  denops
+    .dispatch("gin", "command", "", ["commit", `--fixup=${target}`])
+    .catch((e) => console.error(`failed to execute git commit: ${e}`));
```
Comment (lambdalisue) [inline denops/gin/action/fixup.ts:56]: I think this technique is required only for `fixup:reword` right? If so, please add `doFixupReword` and use this only on that function and implement `doFixup` with `await`.

Comment (lambdalisue) [reply denops/gin/action/fixup.ts]: Then how about `doFixupInteractive`?

Comment (atusy) [reply denops/gin/action/fixup.ts]: Sounds good

Comment (atusy) [reply denops/gin/action/fixup.ts]: done

**Pull request merged**